File: SDKs/amazonaws.com/pinpoint/2016-12-01/java/lib/src/main/java/openapisdk/models/shared/RecencyDimension.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * RecencyDimension
 * Specifies criteria for including or excluding endpoints from a segment based on how recently an endpoint was active.
**/
public class RecencyDimension {
    @JsonProperty("Duration")
    public DurationEnum duration;
    public RecencyDimension withDuration(DurationEnum duration) {
        this.duration = duration;
        return this;
    }
    @JsonProperty("RecencyType")
    public RecencyTypeEnum recencyType;
    public RecencyDimension withRecencyType(RecencyTypeEnum recencyType) {
        this.recencyType = recencyType;
        return this;
    }
}

File: SDKs/amazonaws.com/pinpoint/2016-12-01/python/src/sdk/models/shared/recencydimension.py
import dataclasses
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import duration_enum as shared_duration_enum
from ..shared import recencytype_enum as shared_recencytype_enum


@dataclass_json
@dataclasses.dataclass
class RecencyDimension:
    r"""RecencyDimension
    Specifies criteria for including or excluding endpoints from a segment based on how recently an endpoint was active.
    """
    
    duration: shared_duration_enum.DurationEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Duration') }})
    recency_type: shared_recencytype_enum.RecencyTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('RecencyType') }})
    
